src/server/tran_server.cpp

```diff
@@ -435,6 +453,7 @@ tran_server::uses_remote_storage () const
 void
 tran_server::connection_handler::set_connection (cubcomm::channel &&chn)
 {
+  auto lockg_conn = std::lock_guard<std::shared_mutex> { m_conn_mtx };
```

Comment:
What's the point to move the lock here? Should remain just before the `assert`.

Reply:
I thought that this location of lock says that the whole job in this function needs the lock acquired, and the lock scope is the same.

Reply:
fair enough

Reply:
modified as you said, though. :)))